astropy/coordinates/baseframe.py

```diff
@@ -1760,7 +1760,7 @@ def separation(
         Computes on-sky separation between this coordinate and another.
 
         For more on how to use this (and related) functionality, see the
-        examples in :doc:`astropy:/coordinates/matchsep`.
+        examples in :ref:`astropy-coordinates-separations-matching`.
```

Comment:
@nstarman , any reason why `:doc:` is preferred over `:ref:`?

Reply:
Ooops, sorry, pinged the wrong person: @eerovaher from https://github.com/astropy/astropy/pull/15677

Reply:
I think for downstream.
If you inherit a doc from an Astropy class that has an intersphinx link it should still work.
If it's an internal ref then it will break.
Intersphinx allows us to have public docstrings that reference the docs.

Reply:
OK thanks for the clarification!

Reply:
It's a hard problem to crack because https://github.com/sphinx-doc/sphinx/issues/9169 is not resolved (https://github.com/sphinx-doc/sphinx/pull/10127 remains unmerged).